lib/passageidentity/auth.rb

```diff
@@ -10,14 +10,124 @@ module Passage
+        raise PassageError.new(message: 'no authentication token in header') unless headers.key?('Authorization')
+
+        @token = headers['Authorization'].split(' ').last
+      end
+
+      # authenticate the token
+      return authenticate_token(@token) if @token
+
+      raise PassageError.new(message: 'no authentication token')
+    end
+
+    def validate_jwt(token)
+      return authenticate_token(token) if token
+
+      raise PassageError.new(message: 'no authentication token')
+    end
+
+    def revoke_user_refresh_tokens(user_id)
+      # rubocop:disable Layout/LineLength
+      warn '[DEPRECATION] `auth.revoke_user_refresh_tokens()` is deprecated. Please use `user.revoke_user_refresh_tokens()` instead.'
+      # rubocop:enable Layout/LineLength
+
+      client = OpenapiClient::TokensApi.new
+      client.revoke_user_refresh_tokens(@app_id, user_id)
+      true
+    rescue Faraday::Error => e
+      raise PassageError.new(
+        message: "failed to revoke user's refresh tokens",
+        status_code: e.response[:status],
+        body: e.response[:body]
+      )
+    end
+
+    # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity, Metrics/ParameterLists
+    def create_magic_link(
+      user_id: '',
+      email: '',
+      phone: '',
+      channel: '',
+      send: false,
+      magic_link_path: '',
+      redirect_url: '',
+      language: '',
+      ttl: 60,
+      type: 'login'
+    )
+      magic_link_req = {}
+      magic_link_req['user_id'] = user_id unless user_id.empty?
+      magic_link_req['email'] = email unless email.empty?
+      magic_link_req['phone'] = phone unless phone.empty?
+
+      # check to see if the channel specified is valid before sending it off to the server
+      unless [PHONE_CHANNEL, EMAIL_CHANNEL].include? channel
+        raise PassageError.new(
+          message:
+            'channel: must be either Passage::EMAIL_CHANNEL or Passage::PHONE_CHANNEL'
+        )
+      end
+      magic_link_req['channel'] = channel unless channel.empty?
+      magic_link_req['send'] = send
+      unless magic_link_path.empty?
+        magic_link_req[
+          'magic_link_path'
+        ] = magic_link_path
+      end
+      magic_link_req['redirect_url'] = redirect_url unless redirect_url.empty?
+      magic_link_req['language'] = language
+      magic_link_req['ttl'] = ttl unless ttl.zero?
+      magic_link_req['type'] = type
+
+      begin
+        gemspec = File.join(__dir__, '../../passageidentity.gemspec')
+        Gem::Specification.load(gemspec)
+        header_params = { 'Passage-Version' => "passage-ruby #{Passage::VERSION}" }
+        header_params['Authorization'] = "Bearer #{@api_key}" if @api_key != ''
+
+        opts = {}
+        opts[:header_params] = header_params
+        opts[:debug_auth_names] = ['header']
+
+        client = OpenapiClient::MagicLinksApi.new
+        client.create_magic_link(@app_id, magic_link_req, opts).magic_link
```

Comment:
i know this wasn't modified in this PR besides moving it, but this header config could be more easily found if setup in the constructor instead like in the `User` class

Reply:
Yeah, it is probably a good time to reduce some of the complexity instead of just silencing the rules. The auth class needs some cleanup as a whole. I was going to follow up with a different PR and fix some things in here. For instance, the Passage version header isn't included in any of the existing auth functions.

Reply:
> follow up with a different PR

sounds good 👍